tools/CouchDB_Import_MRI.php

```diff
@@ -199,6 +229,110 @@ function _addMRIFeedback($current_feedback, $scan_type, $mri_feedback)
     }
 
     /**
+    * Gets the date according to the type
+    *
+    * @param MRIFile $file  file object
+    * @param string  $type  type of the date
+    * @param array   $array array containing the date
+    *
+    * @return date if exists, if not an empty string
+    */
+    function _getDate($file, $type, $array)
+    {
+        if (preg_match(
+            "/(\d{4})-?(\d{2})-?(\d{2})/", 
+            $file->getParameter($type), $array
+        )) {
+            return (mktime(12, 0, 0, $array[2], $array[3], $array[1]));
+        } else {
+            return "";
+        }
+    }
+
+    /**
+    * Gets a rejected parameter according to its type
+    *
+    * @param MRIFile $file  file object
+    * @param string  $type  type of the rejected
+    * @param array   $array array containing rejected
+    *
+    * @return parameter of the rejected
+    */
+    function _getRejected($file, $type, $array)
+    {
+        $parameter = 'processing:' . $type . '_rejected';
+        if (preg_match(
+            "/(Directions)([^\(]+)(\(\d+\))/",
+            $file->getParameter($parameter), $array
+        )) {
+            $dirList = preg_split('/\,/', $array[2]);
+            if (count($dirList) > 1) {
+                sort($dirList);
+            }
```

Comment:
This code is a little obtuse. What's this trying to accomplish that just directly returning getParameter wouldn't? What/why is it sorting? There should be a comment explaining what's going on here.